src/sdk/nsfs_glacier_backend/eeadm.js

```diff
@@ -0,0 +1,409 @@
+        })
+        .filter(Boolean);
+}
+
+/**
+ * migrate takes name of a file which contains the list
+ * of the files to be migrated to tape.
+ * 
+ * The file should be in the following format: <filename>
+ * 
+ * The function returns the names of the files which failed
+ * to migrate.
+ * @param {string} file filename
+ * @returns {Promise<string[]>} failedfiles
+ */
+async function migrate(file) {
+    try {
+        await asyncexec(`${path.join(config.NSFS_GLACIER_EEADM_BIN_DIR, 'eemigrate')} ${file}`);
+        return [];
+    } catch (error) {
+        return eeadm_failure_handler(error);
+    }
+}
+
+/**
+ * recall takes name of a file which contains the list
+ * of the files to be recall to tape.
+ * 
+ * The file should be in the following format: <filename>
+ * 
+ * The function returns the names of the files which failed
+ * to recall.
+ * @param {string} file filename
+ * @returns {Promise<string[]>} failed files
+ */
+async function recall(file) {
+    try {
+        await asyncexec(`${path.join(config.NSFS_GLACIER_EEADM_BIN_DIR, 'eerecall')} ${file}`);
+        return [];
+    } catch (error) {
+        return eeadm_failure_handler(error);
+    }
+}
+
+/**
+ * mmapplypolicy takes a policy name and runs passes it onto `mmapplypolicy` command. The engine
+ * will only generate the list of candidates and will store it in the given destination.
+ * 
+ * NOTE: `config.NSFS_GLACIER_EEADM_DEV_DIR` is used for device directory
+ * @param {string} file
+ * @param {string} dest
+ */
+async function mmapplypolicy(file, dest) {
+    try {
+        await asyncexec(`mmapplypolicy ${config.NSFS_GLACIER_EEADM_DEV_DIR} -P ${file} -I defer -f ${dest}`);
+    } catch (error) {
+        console.error('unexpected error occured while executing mmapplypolicy', error);
+
+        // The generated files are supposed to be consumed by another function, throw
+        // an error here to indicate failure.
+        throw error;
+    }
+}
+
+/**
+ * generate_expiration_policy takes a bucket_root path and generates a policy
+ * file which can be consumed by spectrum scale policy engine
+ * @param {string} bucket_root 
+ * @returns {string}
+ */
+function generate_expiration_policy(bucket_root) {
+    return `
+define(user_exclude_list,(PATH_NAME LIKE '/ibm/gpfs/.ltfsee/%' OR PATH_NAME LIKE '/ibm/gpfs/.SpaceMan/%'))
+define(is_premigrated,(MISC_ATTRIBUTES LIKE '%M%' AND MISC_ATTRIBUTES NOT LIKE '%V%'))
+
+RULE 'SYSTEM_POOL_PLACEMENT_RULE' SET POOL 'system'
+
+RULE EXTERNAL POOL 'ltfs'
+EXEC '${path.join(config.NSFS_GLACIER_EEADM_BIN_DIR, 'eemigrate')}'
+
+RULE 'LTFS_EE_FILES' MIGRATE FROM POOL 'system'
+TO POOL 'ltfs'
+WHERE
+    FILE_SIZE > 0
+    AND PATH_NAME LIKE '${bucket_root}/%'
+    AND xattr('user.storage_class') = 'GLACIER'
+    AND (
+            is_premigrated
+            AND
+            CURRENT_TIMESTAMP - TIMESTAMP(SUBSTR(xattr('${GlacierBackend.XATTR_RESTORE_EXPIRY}'), 0, 10)) >= INTERVAL '0' DAYS
+    )
+    AND NOT user_exclude_list
```

Comment:
consider extracting this from our code into a script in NSFS_GLACIER_EEADM_BIN_DIR.
the script can take the arguments that we want to send, like expiry / bucket_name / etc and our code will not need to maintain.